utils/events.py

```diff
@@ -135,3 +151,307 @@ def ensure_event_happens(self, target_type, target_id, event_type):
         if len(events) == 0:
             raise AssertionError(
                 'Event {}/{}/{} did not happen.'.format(event_type, target_type, target_id))
+
+
+class EventBuilder(object):
+    """
+    this class just simplifies "expected" event creation.
+
+    Usage:
+        builder = EventBuilder(appliance)
+        evt = builder.new_event(target_type='VmOrTemplate',
+                                target_name='my_lovely_vm',
+                                event_type='vm_create')
+    """
+    def __init__(self, appliance):
+        self._tool = EventTool(appliance)
+
+    def new_event(self, *attrs, **kwattrs):
+        event = Event(event_tool=self._tool)
+        for name, value in kwattrs.items():
+            event.add_attrs(EventAttr(**{name: value}))
+
+        for attr in attrs:
+            event.add_attrs(EventAttr(**attr))
+        return event
+
+
+class EventAttr(object):
+    """
+    contains one event attribute and the method for comparing it.
+    """
+    def __init__(self, attr_type=None, cmp_func=None, **attrs):
+        if len(attrs) > 1:
+            raise ValueError('event attribute can have only one key=value pair')
+
+        self.name, self.value = attrs.items()[0]
+        self.type = attr_type or type(self.value)
+        self.cmp_func = cmp_func
```

Comment:
i missed this one before, but cmp_func is a completely miss-leading name, as its a matcher, not a compare

i believe its fine to merge, but we should take a intense look on if we want to rename it to something more fitting later